File: rx-codelab/src/main/java/com/jraska/rx/codelab/http/RequestInfoCache.java
package com.jraska.rx.codelab.http;

import io.reactivex.Observable;

import java.util.concurrent.TimeUnit;

/**
 * Showcase purpose - not a cache at all :P
 */
public class RequestInfoCache {
  public static Observable<RequestInfo> getRequestInfo() {
    return Observable.fromCallable(() -> getRequestSync())
      .delay(400, TimeUnit.MILLISECONDS);
  }

  private static RequestInfo getRequestSync() {
    RequestInfo requestInfo = new RequestInfo();
    requestInfo.origin = "cache";

    return requestInfo;
  }
}

File: rx-codelab/src/main/java/com/jraska/rx/codelab/http/RequestInfoCache.kt
package com.jraska.rx.codelab.http

import io.reactivex.Observable

import java.util.concurrent.TimeUnit

/**
 * Showcase purpose - not a cache at all :P
 */
object RequestInfoCache {
  val requestInfo: Observable<RequestInfo>
    get() = Observable.fromCallable { requestSync }
      .delay(400, TimeUnit.MILLISECONDS)

  private val requestSync: RequestInfo
    get() {
      val requestInfo = RequestInfo()
      requestInfo.origin = "cache"

      return requestInfo
    }
}
